pep-0698.rst

```diff
@@ -447,8 +447,9 @@ We decided against it because:
   be considerable benefits.
 - We believe that it would be rarely used and catch relatively few bugs.
 
-  - The author of the ``overrides`` package
-    `has noted <https://discuss.python.org/t/pep-698-a-typing-override-decorator/20839/4>`_
+  - The author of the
+    `Overrides package <https://pypi.org/project/overrides/>`_ has
```

Comment:
Here, you _could_ reuse the target from above, but it runs the risk of breaking the link if the other reference is removed or changed, so best to just keep it as is, I think.

Reply:
:+1:

Reply:
If we don't plan to re-use the reference name elsewhere, would it be better to make it anonymous so as to avoid cluttering the namespace?

How are name collisions handled by Sphinx?

It might not matter as much for the PEPs repo; but, could it lead to incorrect references in a larger docset?

Reply:
> If we don't plan to re-use the reference name elsewhere, would it be better to make it anonymous so as to avoid cluttering the namespace?

The way I view it, if the URL is more or less pseudo-canonical for the link text in the context (i.e., we would be unlikely to link `Overrides package` to anything else in this PEP), then there's no harm in omitting the `_` just in case.

In any case, given this is perhaps one of the final significant edits to an accepted PEP, this is a bit of just counting angels on pinheads. :)

> How are name collisions handled by Sphinx?

Technically speaking, hyperlink targets are a feature of baseline reST and processed by Docutils, not Sphinx. Thus, they are namespaced per-document, and accordingly do not depend on the number of documents present (which would actually be a much bigger problem here than most places, otherwise).

Defining or attempting to use a duplicate hyperlink target produces loud warnings, and will cause the build to exit if `-W` is passed (which we're working toward by default on this repo, and already do on CPython, Devguide, etc), or if the docutils errorlevel is set (as we do here).

For example, if a document has:

```rst
`test link <https://example.com>`_

`test link`_

`test link <https://example.org>`_

`test link`_
```

the build will output:

```
/Path/To/FIle.rst:2: WARNING: Duplicate explicit target name: "test link".
/Path/To/File.rst:13: ERROR: Duplicate target name, cannot be used as a unique reference: "test link".
/Path/To/File.rst:17: ERROR: Duplicate target name, cannot be used as a unique reference: "test link".
```

Reply:
Cool, thanks very much for the explainer!